src/frontends/tensorflow/src/op/max_pool.cpp

```diff
@@ -14,56 +15,103 @@ namespace frontend {
+                                 "MaxPool3D data format is neither NDHWC nor NCDHW");
+        is_nhwc = (tf_data_format == "NDHWC");
+    }
+
+    // prepare attributes for OpenVINO MaxPool operation
+    ov::Strides strides(spatial_dims_num);
+    ov::Strides dilations = (spatial_dims_num == 2 ? ov::Strides({1, 1}) : ov::Strides({1, 1, 1}));
+    ov::Shape kernel_sizes(spatial_dims_num);
+    ov::frontend::tensorflow::convert_nhwc_to_hw(is_nhwc, tf_strides, strides);
+    ov::frontend::tensorflow::convert_nhwc_to_hw(is_nhwc, tf_kernel_sizes, kernel_sizes);
+
+    ov::CoordinateDiff pads_begin;
+    ov::CoordinateDiff pads_end;
+    if (auto_pad == ov::op::PadType::EXPLICIT) {
+        fill_explicit_pads_vectors(node, is_nhwc, spatial_dims_num, tf_explicit_paddings, pads_begin, pads_end);
+    }
+
+    // prepare input to MaxPool
+    convert_nhwc_to_nchw(is_nhwc, input);
 
-    bool is_nhwc = (tf_data_format == "NHWC") || (tf_data_format == "NDHWC");
+    auto max_pool_node = std::make_shared<ov::opset8::MaxPool>(input,
+                                                               strides,
+                                                               dilations,
+                                                               ov::Shape(pads_begin.begin(), pads_begin.end()),
+                                                               ov::Shape(pads_end.begin(), pads_end.end()),
+                                                               kernel_sizes,
+                                                               ov::op::RoundingType::FLOOR,
+                                                               auto_pad);
+    auto max_pool = max_pool_node->output(0);
+    ov::frontend::tensorflow::convert_nchw_to_nhwc(is_nhwc, max_pool);
+    ov::frontend::tensorflow::set_node_name(node.get_name(), max_pool.get_node_shared_ptr());
+    return {max_pool};
+}
+
+OutputVector translate_max_pool(const NodeContext& node, size_t spatial_dims_num) {
+    // MaxPool2D and MaxPool3D have ksize and strides as attributes
+    // retrieve attributes
+    auto strides = node.get_attribute<std::vector<int64_t>>("strides");
+    auto kernel_sizes = node.get_attribute<std::vector<int64_t>>("ksize");
+    return translate_max_pool_util(node, spatial_dims_num, kernel_sizes, strides);
+}
+
+OutputVector translate_max_pool_v2(const NodeContext& node) {
+    // MaxPoolV2 has ksize and strides as input parameters
+    TENSORFLOW_OP_VALIDATION(node, node.get_input_size() > 2, "MaxPoolV2 operation must have at least three inputs.");
+    auto ksize = node.get_input(1);
+    auto strides = node.get_input(2);
+
+    auto ksize_constant = get_constant_from_source(ksize);
+    TENSORFLOW_OP_VALIDATION(node, ksize_constant, "MaxPoolV2 is supported only with constant ksize.");
+    auto strides_constant = get_constant_from_source(strides);
+    TENSORFLOW_OP_VALIDATION(node, ksize_constant, "MaxPoolV2 is supported only with constant strides.");
```

Comment:
```suggestion
    TENSORFLOW_OP_VALIDATION(node, strides_constant , "MaxPoolV2 is supported only with constant strides.");
```

Reply:
Will be fixed in the next PR. It is not a blocker now to merge since CI is too-too slow.